File: apps/api/plane/app/serializers/milestone.py
# Copyright (c) 2023-present Plane Software, Inc. and contributors.
# SPDX-License-Identifier: AGPL-3.0-only
# See the LICENSE file for details.

# Module imports
from .base import BaseSerializer

from plane.db.models import Milestone


class MilestoneSerializer(BaseSerializer):
    class Meta:
        model = Milestone
        fields = "__all__"
        read_only_fields = ["workspace", "project", "created_by", "updated_by", "created_at", "updated_at", "deleted_at"]

File: packages/types/src/milestone/milestone.ts
/**
 * Copyright (c) 2023-present Plane Software, Inc. and contributors
 * SPDX-License-Identifier: AGPL-3.0-only
 * See the LICENSE file for details.
 */

export interface IMilestone {
  id: string;
  project: string;
  workspace: string;
  name: string;
  description: string | null;
  target_date: string | null;
  sort_order: number;
  created_at: string;
  updated_at: string;
  created_by: string;
  updated_by: string;
}

export type TPartialMilestone = Partial<IMilestone>;
